ruby/vampires.rb

```diff
@@ -0,0 +1,70 @@
+print "how many employees will be processed?"
+	employee_number = gets.chomp.to_i
+	current_count = 0
+
+while current_count < employee_number
+
+
+print "What is your name?\n"
+	interviewer_name = gets.chomp
+print "How old are you?\n"
+	age = gets.chomp.to_i
+print "What year were you born?\n"
+	dob = gets.chomp.to_i
+print "Our company cafeteria serves garlic bread. Should we order some for you?(y/n)\n"
+	garlic_question = gets.chomp
+print "Would you like to enroll in the company's health insurance(y/n)\n"
+	health_insurance = gets.chomp
+print "please list any allergies you may have, one at a time type in done when finished.\n"
+valid_input = false
+until valid_input
+	allergies = gets.chomp
+	if allergies == "done"
+		valid_input = true
+	elsif allergies == "sunshine"
+		valid_input = true
+	else
+	end
+end
+
+current_year = Time.new.year
+age_match = current_year - dob
+if age_match == age
+	age_test = true	
+else
+	age_test = false
+end
+
+if garlic_question == "y"
+	likes_garlic = true
+else
+	likes_garlic = false
+end
+
+if health_insurance == "y"
+	wants_health_insurance = true
+else
+	wants_health_insurance = false
+end
+
+if interviewer_name == "Drake Cula" || interviewer_name == "Tu Fang" 

```

Comment:
Let consider order of operations here. I understand that the instructions may be guiding you to just put in conditions as they are mentioned. However, we could clean this up and make the file much more efficient by analyzing our conditions for optimal calls and runtime.

If the employee is named "Drake Cula" or "Tu Fang" then we know that they are definitely a vampire. If that's the case, is there any need to even run the rest of the code?

Consider nesting your conditions and creating a strong order of operations for optimized logic and runtime.
```
if name == "Drake Cula" || name == "Tu Fang"
   "definitely a vampire"
else
  if age_check
    do something
  elsif garlic_and_insurance_check
    do something
  else
    do something
  end
end
```